kyuubi-main/src/main/scala/org/apache/kyuubi/session/KyuubiSessionImpl.scala

```diff
@@ -95,7 +104,7 @@ class KyuubiSessionImpl(
         var sh = getServerHost
         val started = System.currentTimeMillis()
         while (sh.isEmpty) {
-          if (process.waitFor(1, TimeUnit.SECONDS)) {
+          if (!isClusterMode && process.waitFor(1, TimeUnit.SECONDS)) {
```

Comment:
does this mean that cluster mode never fails with client exceptions/errors?

Reply:
yes, the engine will start in AM.

Reply:
You did not get my point

Reply:
In client mode, we can use `process.waitFor` to return true to determine engine startup exceptions. But in cluser mode, if   'spark.yarn.submit.waitAppCompletion=false' is set , `process.waitFor` always return true.

Reply:
The client exception is useful. for example, your YARN cluster is crashed, or maybe the server side is configured wrong or the user overwrites some setting, then Kyuubi failed to start the engine proc, we need to capture the err.

Reply:
@yaooqinn I've fixed it.